src/core/thread/link_metrics.cpp

```diff
@@ -44,39 +44,180 @@
+                                                          uint8_t                         aSeriesId,
+                                                          const otLinkMetricsSeriesFlags &aSeriesFlags,
+                                                          const otLinkMetrics *           aLinkMetricsFlags)
 {
-    otError       error = OT_ERROR_NONE;
-    Tlv           tlv;
-    uint8_t       queryId;
-    bool          hasQueryId  = false;
-    uint8_t       length      = 0;
-    uint16_t      startOffset = aMessage.GetLength();
-    uint16_t      offset;
-    uint16_t      endOffset;
-    otLinkMetrics linkMetrics;
+    otError     error = OT_ERROR_NONE;
+    Tlv         forwardProbingRegistrationSubTlv;
+    SeriesFlags seriesFlags;
+    uint8_t     typeIdFlagsOffset =
+        sizeof(Tlv) + sizeof(uint8_t) * 2; // Forward Probing Registration sub-TLV header + SeriesId + SeriesFlags
+    uint8_t subTlvs[sizeof(Tlv) + sizeof(uint8_t) * 2 + sizeof(LinkMetricsTypeIdFlags) * kMaxTypeIdFlags];
+    uint8_t typeIdFlagsCount =
+        aLinkMetricsFlags == nullptr
+            ? 0
+            : TypeIdFlagsFromLinkMetricsFlags(reinterpret_cast<LinkMetricsTypeIdFlags *>(subTlvs + typeIdFlagsOffset),
+                                              *aLinkMetricsFlags); // Directly transform `aLinkMetricsFlags` into
+                                                                   // LinkMetricsTypeIdFlags and put them into `subTlvs`
+
+    VerifyOrExit(aSeriesId > kQueryIdSingleProbe, error = OT_ERROR_INVALID_ARGS);
+
+    forwardProbingRegistrationSubTlv.SetType(kForwardProbingRegistration);
+    forwardProbingRegistrationSubTlv.SetLength(
+        sizeof(uint8_t) * 2 +
+        sizeof(LinkMetricsTypeIdFlags) *
+            typeIdFlagsCount); // SeriesId + SeriesFlags + typeIdFlagsCount * LinkMetricsTypeIdFlags
+
+    seriesFlags.Clear();
+    if (aSeriesFlags.mLinkProbe)
+    {
+        seriesFlags.SetLinkProbeFlag();
+    }
+    if (aSeriesFlags.mMacData)
+    {
+        seriesFlags.SetMacDataFlag();
+    }
+    if (aSeriesFlags.mMacDataRequest)
+    {
+        seriesFlags.SetMacDataRequestFlag();
+    }
+    if (aSeriesFlags.mMacAck)
+    {
+        seriesFlags.SetMacAckFlag();
+    }
+
+    memcpy(subTlvs, &forwardProbingRegistrationSubTlv, sizeof(forwardProbingRegistrationSubTlv));
+    memcpy(subTlvs + sizeof(forwardProbingRegistrationSubTlv), &aSeriesId, sizeof(aSeriesId));
+    memcpy(subTlvs + sizeof(forwardProbingRegistrationSubTlv) + sizeof(aSeriesId), &aSeriesFlags, sizeof(aSeriesFlags));
+
+    error = Get<Mle::MleRouter>().SendLinkMetricsManagementRequest(aDestination, subTlvs,
+                                                                   forwardProbingRegistrationSubTlv.GetSize());
 
-    memset(&linkMetrics, 0, sizeof(linkMetrics));
+exit:
+    otLogDebgMle("SendMgmtRequestForwardTrackingSeries, error:%s", otThreadErrorToString(error));
```

Comment:
> should here be match = mSeriesFlags.IsMacDataFlagSet() || mSeriesFlags.IsLinkProbeFlagSet()?

If `IsMacDataFlagSet` is `true`, this message would be accounted as a mac data frame. `VerifyOrExit(!mSeriesFlags.IsMacDataFlagSet());` is used to avoid duplication. 

For example, when `IsMacDataFlagSet` is true and we receive a Link Probe message. The Link Probe message would be accounted first at `AggregateLinkMetrics` in `Mac::HandleReceivedFrame`. And then accounted again at `AggregateLinkMetrics` in `Mle::HandleLinkProbe`. I did this because Link Probe are not link layer messages so I cannot directly count it in `Mac`. If `IsMacDataFlagSet` is true, Link Probe would be accounted in Mac layer and won't be accounted in Mle. If `IsMacDataFlagSet` is false and `IsLinkProbeFlagSet()` is true, Link Probe would be accounted in Mle.